lib/ukschedcoop/schedcoop.c

```diff
@@ -70,186 +71,212 @@ static void schedcoop_schedule(struct uk_sched *s)
 		UK_CRASH("Must not call %s from a callback\n", __func__);
 #endif
 
-	do {
-		/* Examine all threads.
-		 * Find a runnable thread, but also wake up expired ones and
-		 * find the time when the next timeout expires, else use
-		 * 10 seconds.
-		 */
-		__snsec now = ukplat_monotonic_clock();
-		__snsec min_wakeup_time = now + ukarch_time_sec_to_nsec(10);
-
-		/* wake some sleeping threads */
-		UK_TAILQ_FOREACH_SAFE(thread, &prv->sleeping_threads,
-				      thread_list, tmp) {
+	/* Examine all sleeping threads.
+	 * Wake up expired ones and find the time when the next timeout expires.
+	 */
+	now = ukplat_monotonic_clock();
+	min_wakeup_time = 0;
 
-			if (thread->wakeup_time && thread->wakeup_time <= now)
+	UK_TAILQ_FOREACH_SAFE(thread, &c->sleep_queue,
+			      queue, tmp) {
+		if (likely(thread->wakeup_time)) {
+			if (thread->wakeup_time <= now)
 				uk_thread_wake(thread);
-
-			else if (thread->wakeup_time < min_wakeup_time)
+			else if (!min_wakeup_time
```

Comment:
This Codacy warning should be ignores. `min_wakeup_time` is `0` only the first time. Then it will be set to `thread->wakeup_time`.